src/test_workflow/test_cluster.py

```diff
@@ -22,25 +41,39 @@ def create(cls, *args):
         """
         cluster = cls(*args)
         try:
-            cluster.create_cluster()
+            cluster.start()
             yield cluster.endpoint(), cluster.port()
         finally:
             cluster.destroy()
 
-    @abc.abstractmethod
-    def destroy(self, test_recorder):
-        """
-        Tear down the cluster and record server logs. If the cluster is already destroyed or has not yet been created then this is a no-op.
-        :param test_recorder: The test recorder to register server logs with.
-        """
-        pass
+    def start():
+        os.makedirs(self.work_dir, exist_ok=True)
+
+        for service in self.services:
+            service.start()
+
+        for service in self.services:
+            service.wait_for_service()
+
+    def terminate(self):
+        self.termination_results = []
+
+        for service in self.services:
+            self.termination_results.append(service.terminate())
+
+        self.__save_test_result_data()
+
+    def __save_test_result_data(self):
+        result = self.termination_results[0]
+
+        test_result_data = TestResultData(
+            self.component_name, self.component_test_config, result.return_code, result.stdout_data, result.stderr_data, result.log_files
+        )
```

Comment:
TestResultData should just take `result` instead of all these things.

Reply:
Since the PR is quite big, I prefer addressing this comment as follow up.